syntax = "proto3";

package mapr;

service WorkerService {
  rpc execute_task (Task) returns (TaskReception) {}
}

message TaskReception {
  string message = 1;
}

message ResultFile {
	string fileName = 1;
}

message TaskCompletion {
  int32 task_id = 1;
  int32 worker_id = 2;
  repeated ResultFile result_files = 3;
}

message FileInfo {
  string fname = 1;
  int32 start = 2;
  int32 end = 3;
}

message Task{
  int32 task_id = 1;
  int32 worker_id = 2;
  int32 num_reducers = 3;
  string task_type = 4;
  repeated FileInfo files = 5;
}
